Assignment2Q2.java

```diff
@@ -0,0 +1,129 @@
+
+public class Assignment2Q2 {
+	public static void main(String[] args) {
+		// construct the tree
+		BinaryTreeNode<Integer> n1 = new BinaryTreeNode<>(1, null, null);
+		BinaryTreeNode<Integer> n2 = new BinaryTreeNode<>(5, null, null);
+		BinaryTreeNode<Integer> n3 = new BinaryTreeNode<>(3, n1, n2);
+		BinaryTreeNode<Integer> n4 = new BinaryTreeNode<>(14, null, null);
+		BinaryTreeNode<Integer> n5 = new BinaryTreeNode<>(9, n3, n4);
+		BinaryTreeNode<Integer> n6 = new BinaryTreeNode<>(19, null, null);
+		BinaryTreeNode<Integer> n7 = new BinaryTreeNode<>(18, null, n6);
+		BinaryTreeNode<Integer> n8 = new BinaryTreeNode<>(16, n5, n7);
+		BinaryTreeNode<Integer> n9 = new BinaryTreeNode<>(186, null, null);
+
+		System.out.println(findCommonAncestors(n9, n1, n8).getData());

```

Comment:
While relying on printed output that has to be verified by human is very useful during development, it typically covers only one scenario and does not scale well.

The code for findCommonAncestors has several edge cases (value not present in the tree, one of the keys being ancestor of the other etc), so it would be a good practice to cover more of them with unit tests. Would you mind adding them, so that when the code is run it gives information about status of different test cases?

You could use JUnit for it or go with simple test method(s) that would execute findCommonAncestors for different scenarios (possibly even using the same tree) and report whether result was as expected or not?